Wrappers/Python/test/test_DataContainer.py

```diff
@@ -781,8 +786,103 @@ def test_reduction_mean(self):
         mean = data.mean()
         expected = numpy.float64(0+1+2+3)/numpy.float64(4)
         numpy.testing.assert_almost_equal(mean, expected)
+
+
+    def directional_reduction_unary_test(self, data, test_func, expected_func, out):
+        # test specifying mean in 1 axis
+        result = test_func(direction=data.dimension_labels[1])
+        numpy.testing.assert_almost_equal(result.as_array(), expected_func(data.as_array(), axis=1))
+        numpy.testing.assert_equal(result.dimension_labels,(data.dimension_labels[0],data.dimension_labels[2]))
+        result = test_func(axis=1)
+        numpy.testing.assert_almost_equal(result, expected_func(data.as_array(), axis=1))
+        # test specifying mean in 2 axes
+        result = test_func(direction=(data.dimension_labels[0],data.dimension_labels[1]))
+        numpy.testing.assert_almost_equal(result.as_array(), expected_func(data.as_array(), axis=(0,1)))
+        numpy.testing.assert_equal(result.dimension_labels,(data.dimension_labels[2],))
+        # test specifying mean in 3 axes
+        result = test_func(direction=(data.dimension_labels[0],data.dimension_labels[1],data.dimension_labels[2]))
+        numpy.testing.assert_almost_equal(result, expected_func(data.as_array()))
+        # test specifying direction with an int   
+        with numpy.testing.assert_raises(ValueError):
+            result = test_func(direction=0)
+        # test providing a DataContainer to out
+        expected_array = expected_func(data.as_array(), axis = 0)
+        test_func(axis=0, out=out)
+        numpy.testing.assert_almost_equal(out.as_array(), expected_array)
+        numpy.testing.assert_equal(out.dimension_labels, (data.dimension_labels[1],data.dimension_labels[2]))
+        test_func(direction=data.dimension_labels[0], out=out)
+        numpy.testing.assert_almost_equal(out.as_array(), expected_array)
+        numpy.testing.assert_equal(out.dimension_labels, (data.dimension_labels[1],data.dimension_labels[2]))
+        # test providing a numpy array to out
+        out_array = numpy.zeros((2,2), dtype=data.dtype)
+        test_func(axis=0, out=out_array)
+        numpy.testing.assert_almost_equal(out_array, expected_array)
+
+    def test_directional_reduction_unary(self):
+        # create DataContainer test class
+        dc =  DataContainer(numpy.array([[[0,1],[2,3]],[[4,5],[6,7]]]), dimension_labels=('vertical', 'horizontal_y', 'horizontal_x'))
+        dc_out = DataContainer(numpy.zeros((2,2)),dimension_labels=('horizontal_y', 'horizontal_x'))
+        # create ImageData test class
+        id = ImageGeometry(2,2,2).allocate(0)
+        id.fill(numpy.array([[[0,1],[2,3]],[[4,5],[6,7]]]))
```

Comment:
create the numpy array once at the top and then reuse it. When you create the data container you can choice to take a copy of the array or a reference to it. I would also use `numpy.array(..., dtype=np.float32)` so we explicitly create the array as our usual float32.